### PR TITLE
feat: assist to merge nested if

### PR DESCRIPTION
**Pull request by l1nxy**
resolve: #16095

### CODE REVIEW

File: crates/ide-assists/src/handlers/merge_nested_if.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    let cond_range = cond.syntax().text_range();
+
+    //check if the then branch is a nested if
+    let then_branch = expr.then_branch()?;
+    let stmt = then_branch.stmt_list()?;
+    if stmt.statements().count() != 0 {
+        return None;
+    }
+
+    let nested_if_to_merge = then_branch.syntax().descendants().find_map(ast::IfExpr::cast)?;
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/merge_nested_if.rs:56]:
```suggestion
    let nested_if_to_merge = then_branch.expr().and_then(|e| match e { ast::Expr::IfExpr(e) => Some(e), _ => None })?;
```

Comment (l1nxy) [reply crates/ide-assists/src/handlers/merge_nested_if.rs:56]: The `then_branch` is of type `BlockExpr`, which does not have an `expr()` method. However, it has a `tail_expr()` method. Should I use it? It also passes all tests.

Comment (Veykril) [reply crates/ide-assists/src/handlers/merge_nested_if.rs:56]: Yes that is the one you want

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Comment by lnicola** [comment]
https://github.com/rust-lang/rust-analyzer/assets/308347/a674b37c-0d77-487e-9431-e6fac30998ad

**Pull request merged**